### PR TITLE
master

### PR DESCRIPTION
**Pull request by larsbrubaker**

### CODE REVIEW

File: GuiAutomation/AutomationRunner.cs [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ public static Task ShowWindowAndExecuteTests(SystemWindow initialSystemWindow, A
 				}));
 
 			// Once either the timeout or the test method has completed, store if a timeout occurred and shutdown the SystemWindow
-			task.ContinueWith((innerTask) =>
+			task.ContinueWith((Action<Task<Task>>)((Task<Task> innerTask) =>
```
Comment (jlewin) [inline GuiAutomation/AutomationRunner.cs]: Should revert compiler generated weirdness back to simplified anonymous type definition

File: agg/SafeList.cs [hunk 2]
```diff
@@ -77,6 +135,11 @@ public void SetParent(T parent)
 			this.parentItem = parent;
 		}
 
+		public ReadOnlyList<T> ReadOnly()
```
Comment (jlewin) [inline agg/SafeList.cs]: Can we describe the benefit and use case for ReadOnly function and how does it differ from SafeList in general?

File: Gui/GUIWidget.cs [hunk 3]
```diff
@@ -1535,55 +1540,65 @@ public virtual void OnChildAdded(EventArgs e)
 
 		public void CloseAllChildren()
 		{
-			while(Children.Count > 0)
+			Children.Modify((Action<List<GuiWidget>>)((List<GuiWidget> list) =>
```
Comment (jlewin) [inline Gui/GUIWidget.cs]: Restore simplified lambda signature

File: Gui/GUIWidget.cs [hunk 4]
```diff
@@ -1492,14 +1491,17 @@ public virtual void Initialize()
 			Initialized = true;
 		}
 
-		public int GetChildIndex(GuiWidget child)
+		public int GetChildIndex(GuiWidget childToFind)
```
Comment (jlewin) [inline Gui/GUIWidget.cs]: Consider if this specialized implementation can be removed in favor of .Children.IndexOf(guiWidget)

File: GuiAutomation/AutomationRunner.cs (hunk 1 above)
Comment (larsbrubaker) [reply GuiAutomation/AutomationRunner.cs]: Fixing

File: agg/SafeList.cs (hunk 2 above)
Comment (larsbrubaker) [reply agg/SafeList.cs]: Removing

File: Gui/GUIWidget.cs (hunk 3 above)
Comment (larsbrubaker) [reply Gui/GUIWidget.cs]: fixed

File: Gui/GUIWidget.cs (hunk 4 above)
Comment (johnlewin) [reply Gui/GUIWidget.cs]: I was under the false impression that SafeList was IList<T>. In the end it looks like the feedback had an effect though

**Pull request merged**